### PR TITLE
Add initial file upload documentation

### PR DESCRIPTION
**Pull request by timja**
Fixes https://github.com/jenkinsci/design-library-plugin/issues/192

See https://github.com/jenkinsci/jenkins/pull/7452

image

<a href="https://gitpod.io/#https://github.com/jenkinsci/design-library-plugin/pull/193"></a>

### CODE REVIEW

File: src/main/webapp/File/file.jelly [hunk 1]
```diff
@@ -0,0 +1 @@
+<f:file field="hello" />
```
Comment (NotMyFault) [inline src/main/webapp/File/file.jelly:1]: Might be worth to ship a default for `accept`, I'd assume that's the most used restriction case.

**Review by NotMyFault — APPROVED** [review]

File: src/main/webapp/File/file.jelly (hunk 1 above)
Comment (timja) [reply src/main/webapp/File/file.jelly:1]: It’s in the docs, accept isn’t used very much, could add another example but not sure if it’s worth it

Comment (NotMyFault) [reply src/main/webapp/File/file.jelly:1]: Oic, can surely leave it as-is then 👍🏻

File: src/main/resources/io/jenkins/plugins/designlibrary/File/index.jelly [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+-->
+
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:s="/lib/samples" xmlns:f="/lib/form">
+  <s:sample>
+    <p class="jdl-paragraph">${%description}</p>
+    <p class="jdl-paragraph">${%parameters}</p>
+    <ul class="jdl-list">
+      <li>id (optional) - ${%parameters.id}</li>
+      <li>accept (optional) - ${%parameters.accept}</li>
```
Comment (timja) [inline src/main/resources/io/jenkins/plugins/designlibrary/File/index.jelly:32]:
```suggestion
```

File: src/main/resources/io/jenkins/plugins/designlibrary/File/index.jelly [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+THE SOFTWARE.
+-->
+
+<?jelly escape-by-default='true'?>
+<j:jelly xmlns:j="jelly:core" xmlns:s="/lib/samples" xmlns:f="/lib/form">
+  <s:sample>
+    <p class="jdl-paragraph">${%description}</p>
+    <p class="jdl-paragraph">${%parameters}</p>
+    <ul class="jdl-list">
+      <li>id (optional) - ${%parameters.id}</li>
```
Comment (timja) [inline src/main/resources/io/jenkins/plugins/designlibrary/File/index.jelly:31]:
```suggestion
```

File: src/main/resources/io/jenkins/plugins/designlibrary/File/index.jelly [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+      <li>accept (optional) - ${%parameters.accept}</li>
+      <li>jsonAware (optional) - ${%parameters.jsonAware}</li>
```
Comment (timja) [inline src/main/resources/io/jenkins/plugins/designlibrary/File/index.jelly:33]:
```suggestion
```

File: src/main/resources/io/jenkins/plugins/designlibrary/File/index.properties [hunk 5]
```diff
@@ -0,0 +1,8 @@
+description=File inputs are simple to use in Jenkins with the Jelly tag.
+parameters=Example parameters:
+parameters.id=Sets an id for the input, useful for UI testing. \
+   Avoid using in repeatable elements as you will end up with duplicate IDs
+parameters.accept=Defines the file types the file input should accept. This string is a comma-separated list.
+parameters.jsonAware=Enables structured form submission.
```
Comment (timja) [inline src/main/resources/io/jenkins/plugins/designlibrary/File/index.properties:6]:
```suggestion
```

**Pull request merged**